rust/blockstore/src/arrow/blockfile.rs

```diff
@@ -682,6 +682,25 @@ impl<'me, K: ArrowReadableKey<'me> + Into<KeyWrapper>, V: ArrowReadableValue<'me
 
         true
     }
+
+    pub async fn get_all_data(&'me self) -> Vec<(&'me str, K, V)> {
+        let block_ids = self.root.sparse_index.get_all_block_ids();
+        let mut result = vec![];
```

Comment:
can preallocate

Reply:
removed it as per suggestion. So can't preallocate what doesn't exist